applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -544,57 +602,57 @@ static void discover_source_cb(struct bt_conn *conn, struct bt_codec *codec, str
 
 	(void)memset(params, 0, sizeof(*params));
 
-	ret = headset_conn_index_get(conn, &conn_index);
+	ret = channel_index_get(conn, NULL, &channel_index);
 
 	if (ret) {
 		LOG_ERR("Unknown connection, should not reach here");
-	} else {
-#if (CONFIG_BT_VCS_CLIENT)
-		ret = ble_vcs_discover(conn, conn_index);
-		if (ret) {
-			LOG_ERR("Could not do VCS discover");
-		}
-#endif /* (CONFIG_BT_VCS_CLIENT) */
+		return;
+	}
 
-		ret = bt_audio_stream_config(conn, &audio_streams[configured_stream_count++],
-					     sources[conn_index].ep, &lc3_preset_nrf5340.codec);
-		if (ret) {
-			LOG_WRN("Could not configure stream");
-			return;
-		}
+	ret = next_free_stream_index_get(&stream_index);
+	if (ret) {
+		LOG_WRN("Failed to get free index");
+		return;
+	}
 
-		if (!group_created) {
-			for (int i = 0; i < ARRAY_SIZE(audio_streams); i++) {
-				group_params[i].stream = &audio_streams[i];
-				group_params[i].qos = &lc3_preset_nrf5340.qos;
-
-				/* Make the first stream be sink */
-				if (i == 0) {
-					group_params[i].dir = BT_AUDIO_DIR_SINK;
-					LOG_DBG("Sink stream: %p", (void *)&audio_streams[i]);
-				} else {
-					group_params[i].dir = BT_AUDIO_DIR_SOURCE;
-					LOG_DBG("Source stream: %p", (void *)&audio_streams[i]);
-				}
-			}
+	ret = bt_audio_stream_config(conn, &audio_streams[stream_index], sources[channel_index].ep,
```

Comment:
channel_index which is defined using the headsets array is used for the sources array index. This does not seem right. This also happens other places in this file, at least for add_remote_sink() and add_remote_source()

Reply:
I see this as a problem if we have a mismatch between the number of sources and CONFIG_BT_MAX_CONN. I added this:
```C
BUILD_ASSERT(ARRAY_SIZE(headsets) >= CONFIG_BT_AUDIO_UNICAST_CLIENT_ASE_SNK_COUNT,
	     "We need to have at least one headset device per ASE SINK");
```
for sinks, should add the same for sources.

This might need some discussing

Reply:
Lets discuss this briefly

Reply:
Discussed this and I will see if we can get rid of audio_sources and audio_sinks all together